crates/bevy_reflect/src/reflect.rs

```diff
@@ -231,6 +238,16 @@ impl Typed for dyn Reflect {
     }
 }
 
+impl TypePath for dyn Reflect {
```

Comment:
This impl seems suspect and prone to causing mistakes? Why is it necessary?

Reply:
I agree, I think we probably want to hold off on this if we can (unless it was in fact necessary for something).